### ISSUE TITLE
Issue: Selecting Objects / Toggling with Alt only works while no objects are selected

### ISSUE THREAD

**Issue by UliAbo** [mapeditor/tiled#3719]
**To Reproduce Issue
(With input from @eishiya from Discord on 2023-05-18):
Holding Alt while selecting toggles between Select Touched Objects and Select Enclosed Objects, but actually trying to drag to select while holding Alt moves the already selected objects. This means toggling with Alt only works while no objects are selected, and if objects are selected, one must use the toolbar toggles. Seems like a bit of an oversight / conflicting modifiers? 

**Specifications:**
-    OS: Win10
-   Tiled Version: 1.10.0

**Comment by bjorn** [comment]
Indeed, Alt also means "move selected objects", so if you want to use it to change the selection mode, you need to press Alt after the mouse button rather than before.

So yes, it's a bit of a conflict, but at least there is a workaround. I would suggest we just leave the behavior as-is, unless you have a suggestion for a better behavior?

**Comment by eishiya** [comment]
> So yes, it's a bit of a conflict, but at least there is a workaround. I would suggest we just leave the behavior as-is, unless you have a suggestion for a better behavior?

I think the current behaviour isn't a big problem though the solution is a bit challenging to discover xP

What makes it confusing and buggy-feeling is that the mode change is *indicated* in the toolbar even when it isn't actually relevant because a move is going to happen or already happening. Is it possible to somehow block Alt from changing the selection mode if it's pressed before clicking and if it would be irrelevant (i.e. when some objects are already selected)? This would be functionally a cosmetic change.

**Comment by bjorn** [comment]
> Is it possible to somehow block Alt from changing the selection mode if it's pressed before clicking and if it would be irrelevant (i.e. when some objects are already selected)? This would be functionally a cosmetic change.

Hmm, yes, that'd be an improvement and it should be possible.

**Comment by kunal649** [comment]
i think the conflict is in `modifiersChanged`. take a look at this : [objectselectiontool.cpp](https://github.com/mapeditor/tiled/blob/eb20ac50b8b3d68f09bdbd4afea99d1d1150cf87/src/tiled/objectselectiontool.cpp#L740) lines 740 - 748. it updates the toolbar unconditionally everytime Alt is pressed, regardless of whether objects are already selected.
and (move-on-alt) logic in mousemoved already check this correctly - > take a look 
[objectselectiontool.cpp](https://github.com/mapeditor/tiled/blob/eb20ac50b8b3d68f09bdbd4afea99d1d1150cf87/src/tiled/objectselectiontool.cpp#L500) 
so when objects are selected and Alt is pressed before clicking, the toolbar toggles to "Select Enclosed" but the subsequent drag will move objects instead. the information is right there , just need to be used in `modifiersChanged` to guard the toolbar update. i tried to implement @eishiya approach in my test branch  : just skip the toolbar update in modifiersChanged  

` if (!mMousePressed && !mapDocument()->selectedObjects().isEmpty() )`

( i play around with code in test branch first before creating a feature branch or PR , look at the vid below ) open to suggestions if i missed something. 

https://github.com/user-attachments/assets/ec60f862-b154-47bc-83ac-bdb596191e78
